tutor/specs/e2e/my-courses.e2e.ts

```diff
@@ -33,4 +33,60 @@ describe('My Courses', () => {
             await page.evaluate(() => document.location.search)
         ).toContain('onboarding=0')
     })
+
+    describe('New Courses UI', () => {
+        it('shows new UI', async () => {
+            await visitPage(page, '/courses')
+            await expect(page).toHaveSelector('testEl=existing-teacher-screen')
+            await expect(page).toHaveSelector('.offering-container')
+            //course card
+            await expect(page).toHaveSelector('.offering-container .course-cards .my-courses-item-wrapper .my-courses-item')
+            //preview card
+            await expect(page).toHaveSelector('.offering-container .course-cards .preview')
+            //create course card
+            await expect(page).toHaveSelector('.offering-container .my-courses-add-zone')
+    
+            //Resources
+            await page.click('testEl=r-e-s-o-u-r-c-e-s-tab')
+            await expect(await page.$eval('#instructor-getting-started', node => node.textContent)).toEqual('Instructor Getting Started Guide')
+            await expect(await page.$eval('#instructor-videos', node => node.textContent)).toEqual(' Video Tutorials ')
+        })
+    
+        it('enters edit mode', async () => {
+            const getOfferingIds = () => {
+                let offeringIds : string[] = []
+                const elements = document.querySelectorAll<HTMLDivElement>('.offering-container')
+                for(let i = 0; i < elements.length; i++){
+                    const element = elements[i];
+                    if(element.dataset.offeringId)
+                        offeringIds.push(element.dataset.offeringId)
+                }
+                return offeringIds;
```

Comment:
@nathanstitt the querySelectorAll returns a `NodeListOf` which does not have a `map` function. Let me try with the lodash `map` and see if it works.

Reply:
@nathanstitt found out that `page.evaluate` is rendered in a completely different browser context so all the modules that are imported, it does not get defined inside the `page.evaluate`. There is a way to add using `page.addScriptTag()`, but my local machine is running of memory when running the test.

Reply:
oh yep, great points.  

To work around missing map, you'll have to convert to a "real" array, either via `Array.from` or `[...]` es7 syntax.  And then to not use `compact` you could use `reduce` and only add the the accumulator if dataset was found.

I spent way to long and came up with this which I think would work:
```js
Array.from(document.querySelectorAll('.offering-container')).reduce((ary, el) => el.dataset?.offeringId 
 ? ary.concat(el.dataset.offeringId) : ary, [])
```

However since you're only wanting elements with a offering-id on dataset, we could add that to the query and then you'd know that dataset.oferingId would be set, like so:
```js
  [...document.querySelectorAll('.offering-container[data-offering-id]')].map(el => el.dataset.offeringId)
```

Reply:
@nathanstitt I was working also with the `Array.from` and it works. Your reduce works much better than mine. I don't think we can use the `...` spread operator. Somehow it is not `defined` inside the `page.evaluate` context.